src/o365/pa/commands/pcf/pcf-init.ts

```diff
@@ -0,0 +1,206 @@
+  namespace: string;
+  name: string;
+  template: string;
+}
+
+/*
+ * Logic extracted from bolt.module.pcf.dll
+ * Version: 0.4.3
+ * Class: bolt.module.pcf.PcfInitVerb
+ */
+class PaPcfInitCommand extends Command {
+  public get name(): string {
+    return commands.PCF_INIT;
+  }
+
+  public get description(): string {
+    return 'Initializes a directory with a new PowerApps component framework project';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.template = args.options.template;
+    return telemetryProps;
+  }
+
+  public action(): CommandAction {
+    const cmd: Command = this;
+    return function (this: CommandInstance, args: CommandArgs, cb: (err?: any) => void) {
+      args = (cmd as any).processArgs(args);
+      (cmd as any).initAction(args, this);
+      cmd.commandAction(this, args, cb);
+    }
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    try {
+      const pcfTemplatePath: string = path.join(__dirname, 'pcf-init', 'assets');
+      const pcfComponentTemplatePath: string = path.join(pcfTemplatePath, 'control', `${args.options.template.toLowerCase()}-template`);
+      const workingDirectory: string = process.cwd();
+      const workingDirectoryName: string = path.basename(workingDirectory);
+      const componentDirectory: string = path.join(workingDirectory, args.options.name);
+      const variables: TemplateVariables = {
+        "$namespaceplaceholder$": args.options.namespace,
+        "$controlnameplaceholder$": args.options.name,
+        "$pcfProjectName$": workingDirectoryName,
+        "pcfprojecttype": workingDirectoryName,
+        "$pcfProjectGuid$": uuidv4()
+      };
+  
+      if (this.verbose) {
+        cmd.log(`name: ${args.options.name}`);
+        cmd.log(`namespace: ${args.options.namespace}`);
+        cmd.log(`template: ${args.options.name}`);
+        cmd.log(`pcfTemplatePath: ${pcfTemplatePath}`);
+        cmd.log(`pcfComponentTemplatePath: ${pcfComponentTemplatePath}`);
+        cmd.log(`workingDirectory: ${workingDirectory}`);
+        cmd.log(`workingDirectoryName: ${workingDirectoryName}`);
+        cmd.log(`componentDirectory: ${componentDirectory}`);
+      }
+  
+      TemplateInstantiator.instantiate(cmd, pcfTemplatePath, workingDirectory, false, variables, this.verbose);
+      TemplateInstantiator.instantiate(cmd, pcfComponentTemplatePath, componentDirectory, true, variables, this.verbose);
+
+      if (this.verbose) {
+        cmd.log(` `);
+      }
+  
+      cmd.log(vorpal.chalk.green(`The PowerApps component framework project was successfully created in '${workingDirectory}'.`));
+      cmd.log(`Be sure to run '${vorpal.chalk.grey('npm install')}' in this directory to install project dependencies.`);
+  
+      cb();
+    }
+    catch (err) {
+      this.handleRejectedODataJsonPromise(err, cmd, cb);
```

Comment:
There are no OData requests in this command, so in the catch we should handle the error message directly.